### PR TITLE
Makes it so player-controller monkeyized humans dont also keep AI control + extra AI controller fixes

### LINKED ISSUE #55544: Active AI despite dead
Reporting client version: 513.1538

## Round ID:
[152221](https://scrubby.melonmesa.com/round/152221)

## Reproduction:

Rust heretic killed a humanized monkey. It still kept moving despite having been killed. Movement was at still normal speed despite being effectively laying on the ground, even if dead.

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
fix: AI controllers now properly check if a mob is client-controlled and dont start processing if its the case.
fix: AI controllers now handle deletion on unpossesion if that's filled in as an argument
fix: Dead monkeys stop attacking things
/:cl:

the second part was requested by timber

closes
#55544

### CODE REVIEW

**Comment by Timberpoes** [comment]
code\datums\ai\_ai_controller.dm:57:error: else: 'else' clause without preceding 'if' statement
code\datums\ai\_ai_controller.dm:49:warning: new_pawn: unused label
tgstation.dmb - 1 error, 1 warning (12/13/20 10:55 pm)

File: code/__DEFINES/ai.dm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 ///Monkey checks
 #define SHOULD_RESIST(source) (source.on_fire || source.buckled || HAS_TRAIT(source, TRAIT_RESTRAINED) || (source.pulledby && source.pulledby.grab_state > GRAB_PASSIVE))
-#define IS_DEAD_OR_INCAP(source) (HAS_TRAIT(source, TRAIT_INCAPACITATED) || HAS_TRAIT(source, TRAIT_HANDS_BLOCKED) || IS_IN_STASIS(source))
+#define IS_DEAD_OR_INCAP(source) (HAS_TRAIT(source, TRAIT_INCAPACITATED) || HAS_TRAIT(source, TRAIT_HANDS_BLOCKED) || IS_IN_STASIS(source) || source.stat)
```
Comment (LemonInTheDark) [inline code/__DEFINES/ai.dm:10]: where is this used exactly?

Comment (Qustinnus) [reply code/__DEFINES/ai.dm:10]: the ai code for monkey all over

Comment (LemonInTheDark) [reply code/__DEFINES/ai.dm:10]: Should it be true if the monkey is in softcrit?

Comment (Qustinnus) [reply code/__DEFINES/ai.dm:10]: this would do that, no?

Comment (LemonInTheDark) [reply code/__DEFINES/ai.dm:10]: Yeah, just wanted to make sure it wasn't being used for "turn off" checks or anything odd, cause I'd think a monkey should be able to move in softcrit

**Pull request merged**